04.ls/ls.rb

```diff
@@ -19,17 +18,31 @@
 PERMISSION_SEPARATOR = 512
 
 def main
-  options = { l: false }
-  opt = OptionParser.new
-  opt.on('-l') { options[:l] = true }
-  argv = opt.parse(ARGV)
+  options = { a: false, r: false, l: false }
+  path = './'
+  ARGV.each do |argv|
+    if FileTest.directory?(argv)
+      path = argv
+    elsif argv.start_with?('-')
+      specified_options = argv.delete('-').chars
+      specified_options.each do |specified_option|
+        case specified_option
+        when 'a' then options[:a] = true
+        when 'r' then options[:r] = true
+        when 'l' then options[:l] = true
+        end
+      end
+    end
+  end
 
-  path = argv[0] || './'
-  FileTest.directory?(path) or return
+  file_names = get_filtered_file_names(path, options)
```

Comment:
`get_xxx_file_names`というメソッドが他にないなら、メソッドを区別する必要がないので

```suggestion
  file_names = get_file_names(path, options)
```

で良いと思います。